staging/src/k8s.io/cri-client/pkg/logs/logs.go

```diff
@@ -452,30 +456,84 @@ func isContainerRunning(ctx context.Context, logger *klog.Logger, id string, r i
 	return true, nil
 }
 
-// waitLogs wait for the next log write. It returns two booleans and an error. The first boolean
+const waitDuration time.Duration = time.Millisecond * 150
+
+type waitLog struct {
+	lastCheck      time.Time
+	lastWriteEvent time.Time
+	// There was a create event.
+	recreated bool
+
+	logger         *klog.Logger
+	id             string
+	logName        string
+	w              *fsnotify.Watcher
+	runtimeService internalapi.RuntimeService
+}
+
+func newWaitLog(logger *klog.Logger, id string, logName string, w *fsnotify.Watcher, runtimeService internalapi.RuntimeService) *waitLog {
+	return &waitLog{
+		logger:         logger,
+		id:             id,
+		logName:        logName,
+		w:              w,
+		runtimeService: runtimeService,
+		recreated:      false,
+	}
+}
+
+// wait for the next log write. It returns two booleans and an error. The first boolean
 // indicates whether a new log is found; the second boolean if the log file was recreated;
 // the error is error happens during waiting new logs.
-func waitLogs(ctx context.Context, logger *klog.Logger, id string, logName string, w *fsnotify.Watcher, runtimeService internalapi.RuntimeService) (bool, bool, error) {
-	// no need to wait if the pod is not running
-	if running, err := isContainerRunning(ctx, logger, id, runtimeService); !running {
-		return false, false, err
+func (wl *waitLog) wait(ctx context.Context) (bool, bool, error) {
+	if time.Since(wl.lastCheck) > waitDuration {
+		// no need to wait if the pod is not running
+		if running, err := isContainerRunning(ctx, wl.logger, wl.id, wl.runtimeService); !running {
+			return false, false, err
+		}
+		wl.lastCheck = time.Now()
+	}
+
+	if wl.recreated {
+		wl.recreated = false
+		// Reopen the log file now.
+		return true, true, nil
```

Comment:
really hard to understand this. recreated is set to false and then we return recreated as `true`